libs/vkd3d/cache.c

```diff
@@ -302,10 +307,29 @@ static HRESULT STDMETHODCALLTYPE d3d12_pipeline_library_SetPrivateData(d3d12_pip
         REFGUID guid, UINT data_size, const void *data)
 {
     struct d3d12_pipeline_library *pipeline_library = impl_from_ID3D12PipelineLibrary(iface);
+    const char *name;
+    HRESULT hr;
 
     TRACE("iface %p, guid %s, data_size %u, data %p.\n", iface, debugstr_guid(guid), data_size, data);
 
-    return vkd3d_set_private_data(&pipeline_library->private_store, guid, data_size, data);
+    if (FAILED(hr = vkd3d_private_data_lock(&pipeline_library->private_store)))
+        return hr;
+
+    if (FAILED(hr = vkd3d_set_private_data(&pipeline_library->private_store, guid, data_size, data)))
```

Comment:
Speaking of _locked, it makes more sense to use vkd3d_set_private_data_locked() here.